tests/integration/suite/daprd/scheduler/grpc/controlplaneerrors.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	apierrors "github.com/dapr/dapr/pkg/api/errors"
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	schedulerv1pb "github.com/dapr/dapr/pkg/proto/scheduler/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/exec"
+	"github.com/dapr/dapr/tests/integration/framework/process/grpc/scheduler"
+	"github.com/dapr/dapr/tests/integration/framework/process/sentry"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(controlPlaneErrors))
```

Comment:
If we are testing errors returned against `scheduler` itself rather than inside `daprd`, consider moving this to `/suite/scheduler/errors`, _or_ create `/suite/daprd/scheduler/grpc/errors/{controlplane.go,app.go}`.

I think I prefer creating a `/grpc/errors` package here.

```suggestion
	suite.Register(new(controlplane))
```

Reply:
I've moved this to `suite/scheduler/api/errors.go` as that made sense to me. I initially moved it to the grpc folder but then realized it wasnt being ran in the integration test suite, so I added the import and was seeing a circular dependency in imports, so I ended up moving it to `suite/scheduler/api/errors.go`.